src/osm-gps-map/layer-gps.c

```diff
@@ -227,49 +227,49 @@ static void _draw(MaepLayerGpsPrivate *priv, cairo_t *cr, OsmGpsMap *map)
-          }
-        else if (priv->compass_mode == COMPASS_MODE_NORTH)
-          {
-            cairo_move_to (cr, -r*cos(priv->compass_azimuth), r*sin(priv->compass_azimuth));
-            cairo_line_to (cr, -4.0*r*sin(priv->compass_azimuth), -4.0*r*cos(priv->compass_azimuth));
-            cairo_line_to (cr, r*cos(priv->compass_azimuth), -r*sin(priv->compass_azimuth));
-            cairo_close_path (cr);
-
-            cairo_set_source_rgba (cr, 0.3, 1.0, 0.3, 0.5);
-            cairo_fill_preserve (cr);
-
-            cairo_set_line_width (cr, 1.0);
-            cairo_set_source_rgba (cr, 0.0, 0.5, 0.0, 0.5);
-            cairo_stroke(cr);
-
-
-            cairo_move_to (cr, r*cos(priv->compass_azimuth), -r*sin(priv->compass_azimuth));
-            cairo_line_to (cr, 4.0*r*sin(priv->compass_azimuth), 4.0*r*cos(priv->compass_azimuth));
-            cairo_line_to (cr, -r*cos(priv->compass_azimuth), r*sin(priv->compass_azimuth));
-            cairo_close_path (cr);
-
-            cairo_set_source_rgba (cr, 1.0, 0.3, 0.3, 0.5);
-            cairo_fill_preserve (cr);
-
-            cairo_set_line_width (cr, 1.0);
-            cairo_set_source_rgba (cr, 0.5, 0.0, 0.0, 0.5);
-            cairo_stroke(cr);
-          }
+      case COMPASS_MODE_OFF:
+        break;
+      case COMPASS_MODE_DEVICE:
+        cairo_move_to (cr, -r*cos(priv->compass_azimuth), -r*sin(priv->compass_azimuth));
+        cairo_line_to (cr, 4.0*r*sin(priv->compass_azimuth), -4.0*r*cos(priv->compass_azimuth));
+        cairo_line_to (cr, r*cos(priv->compass_azimuth), r*sin(priv->compass_azimuth));
+        cairo_close_path (cr);
+
+        cairo_set_source_rgba (cr, 0.0, 1.0, 1.0, 0.5);
+        cairo_fill_preserve (cr);
+
+        cairo_set_line_width (cr, 1.0);
+        cairo_set_source_rgba (cr, 0.0, 0.5, 0.5, 0.5);
+        cairo_stroke(cr);
+        break;
+      case COMPASS_MODE_NORTH:
+        cairo_move_to (cr, -r*cos(priv->compass_azimuth), r*sin(priv->compass_azimuth));
+        cairo_line_to (cr, -4.0*r*sin(priv->compass_azimuth), -4.0*r*cos(priv->compass_azimuth));
+        cairo_line_to (cr, r*cos(priv->compass_azimuth), -r*sin(priv->compass_azimuth));
+        cairo_close_path (cr);
+
+        cairo_set_source_rgba (cr, 0.3, 1.0, 0.3, 0.5);
+        cairo_fill_preserve (cr);
+
+        cairo_set_line_width (cr, 1.0);
+        cairo_set_source_rgba (cr, 0.0, 0.5, 0.0, 0.5);
+        cairo_stroke(cr);
+
+
+        cairo_move_to (cr, r*cos(priv->compass_azimuth), -r*sin(priv->compass_azimuth));
+        cairo_line_to (cr, 4.0*r*sin(priv->compass_azimuth), 4.0*r*cos(priv->compass_azimuth));
+        cairo_line_to (cr, -r*cos(priv->compass_azimuth), r*sin(priv->compass_azimuth));
+        cairo_close_path (cr);
+
+        cairo_set_source_rgba (cr, 1.0, 0.3, 0.3, 0.5);
+        cairo_fill_preserve (cr);
+
+        cairo_set_line_width (cr, 1.0);
+        cairo_set_source_rgba (cr, 0.5, 0.0, 0.0, 0.5);
+        cairo_stroke(cr);
+        break;
       }
```

Comment:
May you add a default case doing break to avoid compiler complain when using full warnings ?